src/Raven.Server/Utils/Cpu/CpuUsageCalculator.cs

```diff
@@ -121,67 +122,36 @@ private double CalculateProcessCpuUsage(ProcessInfo currentInfo, double machineC
             return processCpuUsage;
         }
 
-        public void Dispose()
+        public virtual void Dispose()
         {
         }
     }
 
+    [SupportedOSPlatform("windows")]
     internal sealed class WindowsCpuUsageCalculator : CpuUsageCalculator<WindowsInfo>
     {
-        protected override (double MachineCpuUsage, double? MachineIoWait) CalculateMachineCpuUsage(WindowsInfo windowsInfo)
-        {
-            var systemIdleDiff = windowsInfo.SystemIdleTime - PreviousInfo.SystemIdleTime;
-            var systemKernelDiff = windowsInfo.SystemKernelTime - PreviousInfo.SystemKernelTime;
-            var systemUserDiff = windowsInfo.SystemUserTime - PreviousInfo.SystemUserTime;
-            var sysTotal = systemKernelDiff + systemUserDiff;
-
-            double machineCpuUsage = 0;
-            if (sysTotal > 0)
-            {
-                machineCpuUsage = (sysTotal - systemIdleDiff) * 100.00 / sysTotal;
-            }
+        private readonly PerformanceCounter _cpuCounter;
 
-            return (machineCpuUsage, null);
+        public WindowsCpuUsageCalculator()
+        {
+            _cpuCounter = new PerformanceCounter("Processor Information", "% Processor Time", "_Total");
         }
 
-        protected override WindowsInfo GetProcessInfo()
+        protected override (double MachineCpuUsage, double? MachineIoWait) CalculateMachineCpuUsage(WindowsInfo windowsInfo)
         {
-            var systemIdleTime = new FileTime();
-            var systemKernelTime = new FileTime();
-            var systemUserTime = new FileTime();
-            if (GetSystemTimes(ref systemIdleTime, ref systemKernelTime, ref systemUserTime) == false)
-            {
-                if (Logger.IsInfoEnabled)
-                    Logger.Info("Failure when trying to get GetSystemTimes from Windows, error code was: " + Marshal.GetLastWin32Error());
-                return null;
-            }
-
-            return new WindowsInfo
-            {
-                SystemIdleTime = GetTime(systemIdleTime),
-                SystemKernelTime = GetTime(systemKernelTime),
-                SystemUserTime = GetTime(systemUserTime)
-            };
+            var cpuUsage = _cpuCounter.NextValue();
+            return (cpuUsage, null);
         }
 
-        [return: MarshalAs(UnmanagedType.Bool)]
-        [DllImport("kernel32.dll", SetLastError = true)]
-        internal static extern bool GetSystemTimes(
-            ref FileTime lpIdleTime,
-            ref FileTime lpKernelTime,
-            ref FileTime lpUserTime);
-
-        [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        internal static ulong GetTime(FileTime fileTime)
+        protected override WindowsInfo GetProcessInfo()
         {
-            return ((ulong)fileTime.dwHighDateTime << 32) | fileTime.dwLowDateTime;
+            return WindowsInfo.Instance;
```

Comment:
in tests we run multiple instances of RavenServer, all of them will be using this instance and that will get disposed. Isn't this a problem? Shouldn't this be per server?